### PR TITLE
feat: adding plugin ability for isBlockComment

### LINKED ISSUE #4346: Plugins should be able to define `isBlockComment`
A lot of the core comment printing logic in prettier relies on the ability to identify block comments

https://github.com/prettier/prettier/blob/master/src/main/comments.js#L371

This is currently a private util tailored to JS, but should be abstracted so that plugins could define their own `isBlockComment` functions

https://github.com/prettier/prettier/blob/master/src/common/util.js#L418

### PR DESCRIPTION
**Pull request by mgrip**
Fixes #4346.

### CODE REVIEW

**Comment by j-f1** [comment]
Remaining usages of `isBlockComment` from `privateUtil`:
https://github.com/prettier/prettier/blob/33d0dcf94415208507faa23eb4fb2cddc12f1b6e/src/language-js/printer-estree.js#L1474
https://github.com/prettier/prettier/blob/33d0dcf94415208507faa23eb4fb2cddc12f1b6e/src/language-js/printer-estree.js#L1933
https://github.com/prettier/prettier/blob/33d0dcf94415208507faa23eb4fb2cddc12f1b6e/src/language-js/printer-estree.js#L1957
https://github.com/prettier/prettier/blob/33d0dcf94415208507faa23eb4fb2cddc12f1b6e/src/language-js/printer-estree.js#L5316

This can be removed since it appears to be unused 🔥
https://github.com/prettier/prettier/blob/1f84880fb16282f3d086de71b5ad9b4e3fca8b67/src/common/util.js#L414-L416
This should be moved to `language-js`:
https://github.com/prettier/prettier/blob/1f84880fb16282f3d086de71b5ad9b4e3fca8b67/src/common/util.js#L422-L435

File: src/main/comments.js [hunk 1]
```diff
@@ -368,7 +368,10 @@ function printLeadingComment(commentPath, print, options) {
   if (!contents) {
     return "";
   }
-  const isBlock = privateUtil.isBlockComment(comment);
+  const isBlockComment = options.printer.isBlockComment
+    ? options.printer.isBlockComment
+    : () => false;
+  const isBlock = isBlockComment(comment);
```
Comment (lydell) [inline src/main/comments.js]: What about

```js
const isBlock = options.printer.isBlockComment
  ? options.printer.isBlockComment(comment)
  : false;
```

? Or even:

```js
const isBlock = options.printer.isBlockComment && options.printer.isBlockComment(comment);
```

**Comment by vjeux** [comment]
The idea looks good to me, I'll let @lydell merge it since he's been reviewing the code

**Comment by lydell** [comment]
Nah, I don't really know what this is doing, I just glanced through the diff and saw some (imo) unnecessarily complicated code :)

**Comment by mgrip** [comment]
@j-f1 @lydell updated 👍

**Comment by suchipi** [comment]
Thanks @mgrip!

**Comment by mgrip** [comment]
Np! Thank you guys for being so responsive 😄

**Pull request merged**